packages/language-server/src/plugins/typescript/DocumentSnapshot.ts

```diff
@@ -437,3 +448,42 @@ export class SvelteSnapshotFragment implements SnapshotFragment {
         }
     }
 }
+
+function getChangeRange(oldSnapshot: DocumentSnapshot, currentSnapshot: DocumentSnapshot) {
+    try {
+        if (oldSnapshot.version >= currentSnapshot.version) {
+            return undefined;
+        }
+
+        const oldText = oldSnapshot.getFullText();
+        const currentText = currentSnapshot.getFullText();
+
+        let diffStart = oldText.length - 1;
+        for (let i = 0; i < currentText.length; i++) {
+            if (oldText.charAt(i) !== currentText.charAt(i)) {
```

Comment:
It's probably better to use `charCodeAt`. `chatAt` will likely allocate intermediate strings.